### PR TITLE
[IMP] hr_timesheet: improve generic UX

### PR DESCRIPTION
**Pull request by kcv-odoo**
Purpose of this PR to improve generic UX of timesheet app.

So, in this PR:
- make tree view editable bottom.
- remove extra domain from task_id field in tree view show user
  able to select all non-private tasks.
- add job_title non-store field to add a custom filter on job_title.
- show unit_amount in red in tree view when its value is less
  than 0.

task-2784776

### CODE REVIEW

**Review by xavierbol — COMMENTED** [review]
LGTM, thanks for the changes! 🙂

**Review by xavierbol — APPROVED** [review]

File: addons/hr_timesheet/views/hr_timesheet_views.xml [hunk 1]
```diff
@@ -22,14 +22,14 @@
             <field name="name">account.analytic.line.tree.hr_timesheet</field>
             <field name="model">account.analytic.line</field>
             <field name="arch" type="xml">
-                <tree editable="top" string="Timesheet Activities" sample="1">
+                <tree editable="bottom" string="Timesheet Activities" sample="1">
                     <field name="date"/>
                     <field name="employee_id" invisible="1"/>
                     <field name="project_id" required="1" options="{'no_create_edit': True}"/>
-                    <field name="task_id" optional="show" options="{'no_create_edit': True, 'no_open': True}" widget="task_with_hours" context="{'default_project_id': project_id}" domain="[('project_id', '=', project_id)]"/>
+                    <field name="task_id" optional="show" options="{'no_create_edit': True, 'no_open': True}" widget="task_with_hours" context="{'default_project_id': project_id}"/>
```
Comment (IT-Ideas) [inline addons/hr_timesheet/views/hr_timesheet_views.xml:29]: If the goal is to allow only not private tasks to be selected, shouldn't we keep a domain like `['project_id', '!=', False]`?

Comment (IT-Ideas) [reply addons/hr_timesheet/views/hr_timesheet_views.xml:29]: May I also ask you to rephrase the commit part:
```
- remove extra domain from task_id field in tree view show user
  user able to select all non private tasks.
```
into:
```
- ensure that all non private tasks can be selected in timesheet tree view.
```

**Review by IT-Ideas — COMMENTED** [review]
Thanks 🙂 
Just a question

File: addons/hr_timesheet/views/hr_timesheet_views.xml (hunk 1 above)
Comment (kcv-odoo) [reply addons/hr_timesheet/views/hr_timesheet_views.xml:29]: the domain is already given in the task_id field definition from python [line](https://github.com/odoo/odoo/blob/master/addons/hr_timesheet/models/hr_timesheet.py#L42). Thanks :slightly_smiling_face:

Comment (IT-Ideas) [reply addons/hr_timesheet/views/hr_timesheet_views.xml:29]: > the domain is already given in the task_id field definition from python [line](https://github.com/odoo/odoo/blob/master/addons/hr_timesheet/models/hr_timesheet.py#L42). Thanks 🙂

Ok great 🙂 Thanks!

Comment (IT-Ideas) [reply addons/hr_timesheet/views/hr_timesheet_views.xml:29]: Can you still rephrase the commit as proposed? Thanks!

Comment (ppr-odoo) [reply addons/hr_timesheet/views/hr_timesheet_views.xml:29]: @IT-Ideas We should also remove it from the form view. [Community Form View](https://github.com/odoo/odoo/blame/master/addons/hr_timesheet/views/hr_timesheet_views.xml#L131)

Comment (IT-Ideas) [reply addons/hr_timesheet/views/hr_timesheet_views.xml:29]: > @IT-Ideas We should also remove it from the form view. [Community Form View](https://github.com/odoo/odoo/blame/master/addons/hr_timesheet/views/hr_timesheet_views.xml#L131)

That would make sense, maybe check with @aju-odoo .
Thanks 🙂

**Review by IT-Ideas — APPROVED** [review]
Thanks 🙂 
@robodoo r+

**Comment by innovara** [comment]
And old PR but may I ask why 'make tree view editable bottom.' was considered a UX improvement?

Because it's clearly not, at least on the List view, when you have a long list of lines on a timesheet, which is what lots of professional services companies using timesheets have. If you have say 30 lines, the last thing you want to do is to have to scroll all the way down to put a description every time you add a new line.

Not to mention that it is very inconsistent that after you refresh, the line you just put at the bottom goes at the top. If the new line is loaded with today's date by default, and the view displays in reversed chronological order, then a new line has to go at the top.

So, in terms of usability / efficiency, you want the new line at the top, and in terms of logical order / workflow / presentation (whatever you want to call it) you also want a new line at the top. So both are aligned and were not in need of improvement. Maybe it's an improvement in other views?

**Pull request closed without merging**